gordo_components/server/views/base.py

```diff
@@ -168,7 +182,59 @@ def get(self):
             (X.index > start - self.frequency.delta)
             & (X.index + self.frequency.delta < end)
         ]
-        return self._process_request(context=context, X=X)
+        if y is None:
+            y = X.copy()
+        return self._process_request(context=context, X=X, y=y)
+
+    @staticmethod
+    def dataframe_from_dict(
+        data: dict, tags: typing.List[str], name: str
+    ) -> typing.Union[pd.DataFrame, Response]:
+        """
+        Convert data supported by ``pandas.DataFrame.from_dict`` into a DataFrame
+        and ensure the columns are the ``tags``.
+
+        Parameters
+        ----------
+        df: Union[dict, list]
+            Any data supported by ``pandas.DataFrame.from_dict``
+        tags: List[str]
+            The expected column names for the dataframe. If data supplied does not
+            have column names, but the length of those columns matches the tags, these
+            tags will be used as column names.
+        name: str
+            How to format the ``flask.Response`` message when referring to the data supplied.
```

Comment:
add that `flask.Response` only occurs if there was an error parsing the dict into a DataFrame